agent/src/main.rs

```diff
@@ -171,14 +168,20 @@ async fn main() -> Result<()> {
         );
     }
 
+    let mut roles: Vec<String> = Vec::new();
+    if blob_path.is_some() {
```

Comment:
Doesn't really matter, but if you change this to `if let Some(blob_path) = blob_path {` then you can get rid of the `:?` in the format string and the output will look nicer.